sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSComposeApp.kt

```diff
@@ -117,28 +124,50 @@ fun POSComposeApp(
             }
             composable(route = POSScreen.BalanceInquiryScreen.name) {
                 BalanceInquiryScreen(
-                    onManualEntryButtonClicked = { navController.navigate(POSScreen.ManualPANEntryScreen.name) },
+                    onManualEntryButtonClicked = { navController.navigate(POSScreen.BIManualPANEntryScreen.name) },
                     onSwipeButtonClicked = { /*TODO*/ },
                     onBackButtonClicked = { navController.popBackStack(POSScreen.ActionSelectionScreen.name, inclusive = false) }
                 )
             }
-            composable(route = POSScreen.ManualPANEntryScreen.name) {
+            composable(route = POSScreen.BIManualPANEntryScreen.name) {
                 ManualPANEntryScreen(
                     merchantId = uiState.merchantId,
                     onSubmitButtonClicked = {
                         if (panElement != null) {
                             viewModel.tokenizeEBTCard(panElement as ForagePANEditText, onSuccess = {
                                 if (it?.ref != null) {
                                     Log.i("POSComposeApp", "Successfully tokenized EBT card with ref: $it.ref")
+                                    navController.navigate(POSScreen.BIPINEntryScreen.name)
                                 }
                             })
                         }
                     },
                     onBackButtonClicked = { navController.popBackStack(POSScreen.BalanceInquiryScreen.name, inclusive = false) },
-                    tokenizedCardData = uiState.tokenizationResult?.toString(),
                     withPanElementReference = { panElement = it }
                 )
             }
+            composable(route = POSScreen.BIPINEntryScreen.name) {
+                PINEntryScreen(
```

Comment:
Woah, so you've basically made a reusable `PINEntryScreen` and `PANEntryScreen` components at this point? OMG we're going to FLY through these stories!!

Reply:
Yeah! So the `POSScreen.BIPINEntryScreen.name` part is just defining a PIN Entry route in the BI Flow, but the underlying Composable it's rendering (the `PINEntryScreen`) uses callbacks so it can be generic. We define what happens after tapping "Submit" or "Back" on either of those screens here in the NavHost, so in the Payment flow, for instance, we can re-use the same `PINEntryScreen` composable but give its buttons different behavior.